src/views/product-downloads-view/helpers.ts

```diff
@@ -16,6 +17,78 @@ export interface SortedReleases {
   }
 }
 
+export const generateDefaultPackageManagers = (
+  product: Product
```

Comment:
Oooh [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) is really neat! :til: I think this is a good middle-ground as you said, because we get to keep the benefits of having a lower maintenance function and we also are more explicit while retaining the relationship with the `Product` interface expressed via TypeScript. Added in https://github.com/hashicorp/dev-portal/pull/143/commits/3e05e352d262a7cb6331bda197781ab6ef752a7e!